user/pages/03.commerce2/02.developer-guide/05.payments/04.create-payment-gateway/01.getting-started/docs.md

```diff
@@ -0,0 +1,203 @@
+* For an on-site gateway, use: `Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OnsitePaymentGatewayBase`.
+* For an off-site gateway, use: `Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OffsitePaymentGatewayBase`.
+
+Here is an example of a payment gateway plugin for Commerce QuickPay (an off-site gateway):
+
+```php
+<?php
+
+namespace Drupal\commerce_quickpay\Plugin\Commerce\PaymentGateway;
+
+use Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OffsitePaymentGatewayBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides the QuickPay offsite Checkout payment gateway.
+ *
+ * @CommercePaymentGateway(
+ *   id = "quickpay_redirect_checkout",
+ *   label = @Translation("QuickPay (Redirect to quickpay)"),
+ *   display_label = @Translation("QuickPay"),
+ *    forms = {
+ *     "offsite-payment" = "Drupal\commerce_quickpay\PluginForm\RedirectCheckoutForm",
+ *   },
+ *   payment_method_types = {"credit_card"},
+ *   credit_card_types = {
+ *     "mastercard", "visa",
+ *   },
+ * )
+ */
+class RedirectCheckout extends OffsitePaymentGatewayBase {
+
+}
+```
+
+An important aspect of your plugin class is its ***annotation***, identified by ***@CommercePaymentGateway***. The annotation contains the information Drupal Commerce needs to *discover* your plugin. To learn more about annotations, see the [Drupal 8 documentation on Annotations-base plugins]. For example, if you enable the Commerce QuickPay module (or rebuild caches if the module is already enabled), you will then see *QuickPay (Redirect to quickpay)* as a payment gateway *Plugin* option:
+
+![QuickPay Plugin](../../images/create-payment-gateway-2.png)
+
+Here is the full list of properties for Commerce Payment Gateway annotation properties:
+
+| Property | Description |
+| -------- | ----------- |
+| id       | The plugin ID, a string. |
+| label    | The payment gateway label, a translatable string. |
+| display_label | The display label, a translatable string. |
+| modes    | An array of supported modes, keyed by machine name. If no modes are provided, the default modes are *Test* and *Live* are provided. The mode labels are translatable strings. |
+| forms    | An array of form classes, keyed by operation. For example, the *Manual* payment gateway has as its *forms* property:<br /><code>  forms = {<br />&nbsp;&nbsp;"add-payment" = "Drupal\commerce_payment\PluginForm\ManualPaymentAddForm",<br />&nbsp;&nbsp;"receive-payment" = "Drupal\commerce_payment\PluginForm\PaymentReceiveForm",<br />},</code> |
+ | js_library | The JavaScript library ID. |
+ | payment_type | The payment type used by the payment gateway, a string. If no payment type is provided, the default value is *payment_default*. |
+ | payment_method_types | An array of the payment method types handled by the payment gateway. If no payment method types are provided, a *credit_card* default type is provided. Other types can be things like *paypal*, or *paypal_credit*. |
```

Comment:
We should clarify that the payment method keys are only used for on-site gateways.